src/main/java/core/basesyntax/dao/ManufacturerDaoJdbcImpl.java

```diff
@@ -0,0 +1,128 @@
+                + "WHERE manufacturer_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(getAllManufacturersQuery)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                manufacturers.add(getManufacturer(resultSet).get());
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get any manufacturer", e);
+        }
+        return manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateQuery = "UPDATE manufacturers SET manufacturer_name = ?, "
+                + "manufacturer_country = ? "
+                + "WHERE manufacturer_id = ? AND manufacturer_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(updateQuery)) {
+            preparedStatement.setString(1, manufacturer.getName());
+            preparedStatement.setString(2, manufacturer.getCountry());
+            preparedStatement.setLong(3, manufacturer.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer " + manufacturer, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteByIdQuery = "UPDATE manufacturers SET manufacturer_deleted = true "
+                + "WHERE manufacturer_id = ? AND manufacturer_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(deleteByIdQuery)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacturer by id " + id, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Manufacturer manufacturer) {
+        String deleteByManufacturerQuery = "UPDATE manufacturers SET manufacturer_deleted = true "
+                + "WHERE manufacturer_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(deleteByManufacturerQuery)) {
+            preparedStatement.setLong(1, manufacturer.getId());
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacturer " + manufacturer, e);
+        }
+    }
+
+    private Optional<Manufacturer> getManufacturer(ResultSet resultSet) throws SQLException {
```

Comment:
```suggestion
    private Manufacturer getManufacturer(ResultSet resultSet) throws SQLException {
```
It doesn't make sense to return Optional here, because it can't return empty optional